src/java/org/apache/nutch/segment/SegmentMerger.java

```diff
@@ -384,40 +385,43 @@ public void setConf(Configuration conf) {
   public void close() throws IOException {
   }
 
-  public void configure(JobConf conf) {
+  public void configure(Job job) {
+    Configuration conf = job.getConfiguration();
     setConf(conf);
     if (sliceSize > 0) {
-      sliceSize = sliceSize / conf.getNumReduceTasks();
+      sliceSize = sliceSize / Integer.parseInt(conf.get("mapreduce.map.tasks"));
```

Comment:
This does not look correct at all... we should be attempting to obtain a slice size value by diving an integer whose value is greater than zero by the configured number of Reduce tasks... not by the configured number of Map tasks.